java-frontend/src/main/java/org/sonar/java/se/ProgramPoint.java

```diff
@@ -40,10 +40,13 @@ private ProgramPoint(CFG.Block block, int i) {
   }
 
   ProgramPoint next() {
-    // pointing one element behind is allowed, it represents terminator of the CFG block
-    Preconditions.checkState(i + 1 <= block.elements().size() + 1,
-      "CFG Block has %s elements but PP at %s was requested", block.elements().size(), i + 1);
-    return new ProgramPoint(block, i + 1);
+    int blockSize = block.elements().size();
+    int nextPP = this.i + 1;
+    // pointing one element above block size is allowed, it represents terminator of the CFG block
+    // FIXME when RelationalSV is involved, we can point up to two elements behind, which is fishy
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)